### PR TITLE
meta-phosphor: npcm8xx.bbclass: add SA TIP_FW to support mimic no_tip…

### PR DESCRIPTION
**Pull request by timlee66**
… mode

SA (Stand Alone) is a special TIP_FW for Google.
It is meant to mimic NO_TIP feature on A1 device.

Tested:
* Set TIP_IMAGE = False and SA_TIP_IMAGE = True and build.
* Flash image-u-boot on A1 TIP device and can boot up successfully.
* Set TIP_IMAGE = False and SA_TIP_IMAGE = False and build.
* Flash image-u-boot on A2 NO_TIP device and can boot up successfully.

### CODE REVIEW

**Comment by maxdog988** [comment]
@minimada  please help review this change

File: meta-phosphor/classes/image_types_phosphor_nuvoton_npcm8xx.bbclass [hunk 1]
```diff
@@ -7,10 +7,12 @@ KMT_TIPFW_BB_BINARY = "Kmt_TipFw_BootBlock.bin"
 KMT_TIPFW_BB_BL31_BINARY = "Kmt_TipFw_BootBlock_BL31.bin"
 KMT_TIPFW_BB_BL31_TEE_BINARY = "Kmt_TipFw_BootBlock_BL31_Tee.bin"
 KMT_TIPFW_BB_UBOOT_BINARY = "u-boot.bin.merged"
+SA_KMT_TIPFW_BINARY := "SA_Kmt_TipFwL0.bin"
```
Comment (minimada) [inline meta-phosphor/classes/image_types_phosphor_nuvoton_npcm8xx.bbclass:10]: Do we need set parameter immediately?

File: meta-phosphor/classes/image_types_phosphor_nuvoton_npcm8xx.bbclass [hunk 2]
```diff
@@ -217,10 +220,22 @@ python do_merge_bootloaders() {
...
+            int(d.getVar('SA_ALIGN', True)), int(d.getVar('ALIGN_END', True)))
+
+            os.rename(os.path.join(d.getVar('DEPLOY_DIR_IMAGE', True), '%s' % d.getVar('BB_BL31_TEE_UBOOT_SA_BINARY',True)),
+            os.path.join(d.getVar('DEPLOY_DIR_IMAGE', True), '%s' % d.getVar('KMT_TIPFW_BB_UBOOT_BINARY',True)))
+        else:
+            os.rename(os.path.join(d.getVar('DEPLOY_DIR_IMAGE', True), '%s' % d.getVar('BB_BL31_TEE_UBOOT_BINARY',True)),
+            os.path.join(d.getVar('DEPLOY_DIR_IMAGE', True), '%s' % d.getVar('KMT_TIPFW_BB_UBOOT_BINARY',True)))
 }
 
 do_pad_binary[depends] += " \
```
Comment (minimada) [inline meta-phosphor/classes/image_types_phosphor_nuvoton_npcm8xx.bbclass:237]: So no TIP image both need pad binary?

File: meta-phosphor/classes/image_types_phosphor_nuvoton_npcm8xx.bbclass [hunk 3]
```diff
@@ -217,10 +220,22 @@ python do_merge_bootloaders() {
         os.path.join(d.getVar('DEPLOY_DIR_IMAGE', True), '%s' % d.getVar('UBOOT_HEADER_BINARY',True)),
         os.path.join(d.getVar('DEPLOY_DIR_IMAGE', True), '%s' % d.getVar('BB_BL31_TEE_UBOOT_BINARY',True)),
         int(d.getVar('UBOOT_ALIGN', True)), int(d.getVar('ALIGN_END', True)))
+
```
Comment (minimada) [inline meta-phosphor/classes/image_types_phosphor_nuvoton_npcm8xx.bbclass:223]: Can you explain what you do at this part in commit message?

**Review by minimada — COMMENTED** [review]
please update your comment

File: meta-phosphor/classes/image_types_phosphor_nuvoton_npcm8xx.bbclass (hunk 1 above)
Comment (timlee66) [reply meta-phosphor/classes/image_types_phosphor_nuvoton_npcm8xx.bbclass:10]: Not actually, it's can just use equal

File: meta-phosphor/classes/image_types_phosphor_nuvoton_npcm8xx.bbclass (hunk 3 above)
Comment (timlee66) [reply meta-phosphor/classes/image_types_phosphor_nuvoton_npcm8xx.bbclass:223]: OK, will add more decription.

File: meta-phosphor/classes/image_types_phosphor_nuvoton_npcm8xx.bbclass (hunk 2 above)
Comment (timlee66) [reply meta-phosphor/classes/image_types_phosphor_nuvoton_npcm8xx.bbclass:237]: Yes.

**Pull request merged**